perun-core/src/main/java/cz/metacentrum/perun/core/impl/ExtSourceREMS.java

```
@@ -0,0 +1,157 @@
+		}
+		return subject;
+	}
+
+	@Override
+	public List<Map<String, String>> findSubjectsLogins(String searchString) throws InternalErrorException {
+		List<Map<String, String>> subjects = super.findSubjectsLogins(searchString);
+		return filterNonExistingUsers(subjects);
+	}
+
+	@Override
+	public List<Map<String, String>> findSubjectsLogins(String searchString, int maxResults) throws InternalErrorException {
+		List<Map<String, String>> subjects = super.findSubjectsLogins(searchString, maxResults);
+		return filterNonExistingUsers(subjects);
+	}
+
+	@Override
+	public List<Map<String, String>> getGroupSubjects(Map<String, String> attributes) throws InternalErrorException {
+		List<Map<String, String>> subjects = super.getGroupSubjects(attributes);
+		return filterNonExistingUsers(subjects);
+	}
+
+	/**
+	 * Filters subjects that does not have a corresponding user in Perun by ues REMS
+	 * or by additionalues_1 in format: {extSourceName}|{extSourceClass}|{eppn}|0
+	 *
+	 * @param subjects subjects that will be filtered
+	 * @return List without filtered subjects
+	 * @throws InternalErrorException
+	 */
+	private List<Map<String, String>> filterNonExistingUsers(List<Map<String, String>> subjects) throws InternalErrorException {
+		List<Map<String, String>> existingSubjects = new ArrayList<>();
+
+		for (Map<String, String> subject : subjects) {
+			if(isExistingUser(subject)) {
+				existingSubjects.add(subject);
+			}
+		}
+
+		return existingSubjects;
+	}
+
+	/**
+	 * Checks if the given subject has a corresponding user in Perun by ues REMS
+	 * or by additionalues_1 in format: {extSourceName}|{extSourceClass}|{eppn}|0
+	 *
+	 * @param subject subject
+	 * @return true if the subject has, false otherwise
+	 * @throws InternalErrorException
+	 */
+	private boolean isExistingUser(Map<String, String> subject) throws InternalErrorException {
+		if (subject.isEmpty()) {
+			return false;
+		}
```

Subject can be also null if something really nasty happened in code. I would prefer to throw an exception in both cases - null or emptiness (there should always be at least login of the subject).